homeassistant/components/fronius/sensor.py

```diff
@@ -682,15 +706,21 @@ def _get_entity_value(self) -> Any:
         new_value = self.coordinator.data[self.solar_net_id][
             self.entity_description.key
         ]["value"]
-        return round(new_value, 4) if isinstance(new_value, float) else new_value
+        if new_value is None:
+            return self.entity_description.default_value
+        if isinstance(new_value, float):
+            return round(new_value, 4)
+        return new_value
 
     @callback
     def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
         try:
             self._attr_native_value = self._get_entity_value()
         except KeyError:
-            return
+            if self.entity_description.default_value is None:
+                return
+            self._attr_native_value = self.entity_description.default_value
```

Comment:
Should this not always get set to `None`?
```suggestion
            self._attr_native_value = None
```

Reply:
The intention was to keep the previous behaviour for entities that don't use `default_value` in their entity description. 

An example for values that shouldn't update or set to `None` if omitted from response would be total energy production of an inverter. 
I'm not 100% sure if those values are even omitted when not producing, but Fronius doesn't have a very consistent API across their devices (and especially older ones are hard to find testers for) so I tend to go the safe route here.

I'll add a comment reflecting that 👍